.github/workflows/main.yml

```diff
@@ -452,6 +456,7 @@ jobs:
     - run: cd .github/actions/github-release && npm install --production
     - name: Publish Release
       uses: ./.github/actions/github-release
+      # TODO: it should be fine to remove this guard
       if: github.event_name == 'push' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')) && github.repository == 'bytecodealliance/wasmtime'
```

Comment:
I'm undecided about leaving this guard in. It seems nice to avoid accidentally triggering this in the future, but it's also redundant with the new `if` at the job level.

Reply:
I'll add a note about keeping it because we run these jobs on `release-*` branches.